server/src/test/java/org/opensearch/index/store/remote/utils/TransferManagerTestCase.java

```diff
@@ -320,9 +320,7 @@ public void testRefCount() throws IOException, InterruptedException {
         BlobFetchRequest blobFetchRequest = BlobFetchRequest.builder().fileName(blobname).directory(directory).blobParts(blobParts).build();
         // It will trigger async load
         transferManager.fetchBlobAsync(blobFetchRequest);
-        Thread.sleep(2000);
-        assertEquals(Optional.of(0), Optional.of(fileCache.getRef(blobFetchRequest.getFilePath())));
-        assertNotNull(fileCache.get(blobFetchRequest.getFilePath()));
+        fileCache.get(blobFetchRequest.getFilePath());
```

Comment:
This still doesn't make sense to me. We're starting an async operation right before this. It might complete before this line is executed, or it might not complete until after the test method completes. We should make these test cases deterministic or else we are likely to continue to have flakiness and/or poor coverage.

It's very strange to me that an async API does not return a completablefuture or some other handle that allows you to join on the operation if desired.